### PR TITLE
Send PR-specific env.vars. to TF

### PR DESCRIPTION
**Pull request by lachmanfrantisek**
There will be a variable for commit, branch and url both for PR source and PR target.

Signed-off-by: Frantisek Lachman <flachman@redhat.com>
Fixes: #1379

---

RELEASE NOTES BEGIN
When running tests for the pull-request job, we now expose environment variables for commit hash, branch and URL both for pull-request source and target. In the test environment, you can use the following variables: `PACKIT_SOURCE_SHA`, `PACKIT_TARGET_SHA`, `PACKIT_SOURCE_BRANCH`, `PACKIT_TARGET_BRANCH`, `PACKIT_SOURCE_URL` and `PACKIT_TARGET_URL`. These variables are not set for test runs for releases and branch pushes.
RELEASE NOTES END

### CODE REVIEW

**Review by FrNecas — APPROVED** [review]
LGTM as far as I can tell, nice extension of the variables

**Review by mfocko — APPROVED** [review]
lgtm otherwise

File: packit_service/worker/testing_farm.py [hunk 1]
```diff
@@ -108,12 +108,46 @@ def fmf_ref(self) -> str:
 
         return self.metadata.commit_sha
 
+    @property
+    def source_branch_sha(self) -> Optional[str]:
+        if not self.metadata.pr_id:
+            return None
+        return self.project.get_pr(int(self.metadata.pr_id)).head_commit
```
[the remark this answers is not in the document]
Comment (lachmanfrantisek) [reply packit_service/worker/testing_farm.py:115]: @mfocko Good point. I just sent the refactored version.

**Review by mfocko — APPROVED** [review]

**Comment by mfocko** [comment]
> Build failed (gate pipeline).
> 
>     * [pre-commit ](https://softwarefactory-project.io/zuul/t/packit-service/build/None) : NODE_FAILURE in 0s

I dodged that by just one day :D

**Pull request merged**